File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetscommercepaymentmethod.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsCommercePaymentMethodStatusEnum } from "./codatdatacontractsdatasetscommercepaymentmethodstatusenum";
import { Expose, Transform } from "class-transformer";


export class CodatDataContractsDatasetsCommercePaymentMethod extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "id" })
  id?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "modifiedDate" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  modifiedDate?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "name" })
  name?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "sourceModifiedDate" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  sourceModifiedDate?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "status" })
  status?: CodatDataContractsDatasetsCommercePaymentMethodStatusEnum;
}

File: go-client-sdk/pkg/models/shared/codatdatacontractsdatasetscommercepaymentmethod.go
package shared

import (
	"time"
)

type CodatDataContractsDatasetsCommercePaymentMethod struct {
	ID                 *string                                                    `json:"id,omitempty"`
	ModifiedDate       *time.Time                                                 `json:"modifiedDate,omitempty"`
	Name               *string                                                    `json:"name,omitempty"`
	SourceModifiedDate *time.Time                                                 `json:"sourceModifiedDate,omitempty"`
	Status             *CodatDataContractsDatasetsCommercePaymentMethodStatusEnum `json:"status,omitempty"`
}
